### PR TITLE
fix(object): close built-ins Object test262 gaps

### LINKED ISSUE #847: Bring Object to 100% test262 conformance (~24 fails)
## Summary

Drive the test262 **Object** failures to zero across `built-ins`, `staging`, and `intl402` — these share root causes, so a single fix typically clears the failure in more than one category.

## Why

Reaching ~100% test262 on supported categories is the objective for the conformance milestone. Baseline (main, 2026-06-26): **97.3% overall** (50,877/52,293; 0 timeouts, 0 infra; `language` and `harness` already at 100%). **Object accounts for ~24** of the 1,416 remaining failures.

## Current behavior

~24 Object tests fail. Concentrations: Object static methods and property-descriptor edge cases.

## Expected behavior

Zero Object failures at the pinned suite SHA (`scripts/test262-suite-sha.txt`), with no regressions elsewhere — the PR conformance gate enforces no PASS to non-PASS transitions.

## Scope notes

- Organize the work by failing sub-area (above); they are independently fixable.
- Land the conformance gates first: harness diagnostics (#830), the crash fix (#831), the skip-list (#832).
- Every fix ships with its JS test in the same commit. Parent #820.

### PR DESCRIPTION
**Pull request by frostney**
## Summary

- Fixes the remaining `built-ins/Object/**` test262 failures by aligning `Object.freeze` proxy descriptors, `Object.prototype.toString` fallback branding, arguments-object branding, and writable global constructor reassignment.
- Preserves project-specific immutable globals (`Goccia`, `globalThis`) and host-object toString tags while applying the standard ECMAScript fallback for standard built-ins.
- Addresses review follow-ups by exposing built-in tag fallback through `HasBuiltinTagFallback`, guarding VM global-assignment `TGocciaTypeError` debug locations when bytecode debug info is absent, and centralizing the runtime `Goccia`/`globalThis` property names.
- Merges the latest `origin/main` into the PR branch.
- Closes #847.

## Testing

- [x] Verified no regressions and confirmed the new feature or bugfix in end-to-end JavaScript/TypeScript tests
- [ ] Updated documentation
- [x] Optional: Verified no regressions and confirmed the new feature or bugfix in native Pascal tests (if AST, scope, evaluator, or value types changed)
- [ ] Optional: Verified no benchmark regressions or confirmed benchmark coverage for the change

Validation run:

- `./build.pas testrunner` (hit stale FPC internal error in `Goccia.Values.ClassHelper.pas`; clean retry below passed)
- `./build.pas --clean testrunner`
- `./build/GocciaTestRunner tests` (11079/11079 passed)
- `./build/GocciaTestRunner tests --mode=bytecode` (11079/11079 passed)
- `bun scripts/run_test262_suite.ts --suite-dir /tmp/goccia-test262-020cb740 --categories built-ins --filter 'built-ins/Object/**' --mode=bytecode --jobs=4 --timeout-ms=20000 --output=/tmp/goccia-object-test262-update-pr.json` (3411/3411 passed, 0 failed, 0 wrapper infra)
- `./format.pas --check`
- `git diff --check`

### CODE REVIEW

File: source/units/Goccia.Values.ObjectValue.pas [hunk 1]
```diff
@@ -484,8 +488,40 @@ function TGocciaObjectValue.ObjectPrototypeToString(const AArgs: TGocciaArgument
...
+      Result := CONSTRUCTOR_STRING
+    else if AObject is TGocciaNumberObjectValue then
+      Result := CONSTRUCTOR_NUMBER
+    else if AObject is TGocciaBooleanObjectValue then
+      Result := CONSTRUCTOR_BOOLEAN
+    else
+    begin
+      Result := AObject.ToStringTag;
+      if IsECMAScriptBuiltinObjectTag(Result) then
+        Result := CONSTRUCTOR_OBJECT;
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.Values.ObjectValue.pas]: Fixed in 62f442ca by moving the fallback decision off tag strings and onto a virtual UsesECMAScriptBuiltinTagFallback hook. The default is false for host/custom objects, and only the actual ECMAScript built-in value types opt in, including the private VM generator path. Revalidated built-ins/Object test262: 3411/3411 passed.

File: source/units/Goccia.VM.pas [hunk 2]
```diff
@@ -15392,7 +15392,15 @@       // ES2022 §15.7.14: execute static block closure with this = class
...
             RegisterToValue(FRegisters[A])) then
+          begin
+            if ((GlobalName = 'Goccia') or (GlobalName = 'globalThis')) and
+               (FGlobalScope.ThisValue is TGocciaObjectValue) and
+               TGocciaObjectValue(FGlobalScope.ThisValue).HasProperty(GlobalName) then
+              raise TGocciaTypeError.Create(
+                Format(SErrorAssignToConstant, [GlobalName]),
+                0, 0, '', nil, SSuggestUseLetNotConst);
             ThrowReferenceError(GlobalName + ' is not defined');
+          end;
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.VM.pas:15440]: Fixed in 2b0eda09 by threading Template.DebugInfo.GetLineForPC/GetColumnForPC into both VM constant-assignment TypeError branches instead of hardcoding 0,0. Full local suites and the built-ins/Object test262 slice are green.
